qt/data/PdfExportDialog.ui

```diff
@@ -170,103 +274,46 @@
-         <number>1200</number>
-        </property>
-        <property name="singleStep">
-         <number>10</number>
-        </property>
-        <property name="value">
-         <number>300</number>
-        </property>
-       </widget>
-      </item>
-      <item row="4" column="0">
-       <widget class="QLabel" name="labelCompressionQuality">
-        <property name="enabled">
-         <bool>false</bool>
-        </property>
-        <property name="text">
-         <string>Compression quality:</string>
-        </property>
-       </widget>
-      </item>
-      <item row="2" column="0">
-       <widget class="QLabel" name="labelDpi">
-        <property name="text">
-         <string>DPI:</string>
-        </property>
-       </widget>
-      </item>
-      <item row="0" column="1">
-       <widget class="QComboBox" name="comboBoxImageFormat"/>
-      </item>
-      <item row="4" column="1">
-       <widget class="QSpinBox" name="spinBoxCompressionQuality">
-        <property name="maximum">
-         <number>100</number>
-        </property>
-        <property name="value">
-         <number>90</number>
-        </property>
-       </widget>
-      </item>
-      <item row="3" column="1">
-       <widget class="QComboBox" name="comboBoxImageCompression"/>
-      </item>
+     <property name="maximumSize">
+      <size>
+       <width>393</width>
+       <height>16777215</height>
+      </size>
+     </property>
+     <property name="title">
+      <string>General</string>
+     </property>
+     <layout class="QGridLayout" name="gridLayout_4">
       <item row="0" column="0">
-       <widget class="QLabel" name="labelImageFormat">
-        <property name="text">
-         <string>Format:</string>
-        </property>
-       </widget>
-      </item>
-      <item row="3" column="0">
-       <widget class="QLabel" name="labelImageCompression">
-        <property name="text">
-         <string>Compression:</string>
-        </property>
-       </widget>
-      </item>
-      <item row="1" column="0">
-       <widget class="QLabel" name="labelDithering">
-        <property name="enabled">
-         <bool>false</bool>
-        </property>
+       <widget class="QLabel" name="labelPasswordOpen">
         <property name="text">
-         <string>Dithering algorithm:</string>
+         <string>Password</string>
         </property>
        </widget>
       </item>
-      <item row="1" column="1">
-       <widget class="QComboBox" name="comboBoxDithering">
-        <property name="enabled">
-         <bool>false</bool>
+      <item row="0" column="1">
+       <widget class="QLineEdit" name="lineEditPasswordOpen">
+        <property name="sizePolicy">
+         <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
+          <horstretch>0</horstretch>
+          <verstretch>0</verstretch>
+         </sizepolicy>
+        </property>
+        <property name="echoMode">
+         <enum>QLineEdit::Password</enum>
```

Comment:
Most applications have a second "confirm" field. Up to you whether you want to include it.

Reply:
Hm... Good idea! Thanks.

Reply:
What should we do if password and confirm password are different? show messagebox with warning and after that show PDF export dialog again?

Reply:
I'd just set the background of the inputs to red as soon as the entries do not match and disable the export button. In general it's better to avoid to many dialogs popping up.

Reply:
FYI, the select pages dialog also uses the red background for error reporting, you can take inspiration from [1].

[1] https://github.com/manisandro/gImageReader/blob/master/qt/src/Recognizer.cc#L408

Reply:
Ok, thank you. i will check way with red background.

Reply:
Added support for red background. I have one trouble with disabling "Ok" button. As you know, we have in PDF export dialog QDialogButtonBox. And i don't know how i can disable only "Ok" button. Can you suggest me anything, except using two buttons instead of QDialogButtonBox?

Reply:
You can use

    QDialogButtonBox::button(QDialogButtonBox::Ok)->setEnabled(false)

By the way how about only setting the background red when:
- The user has finished editing the confirm line edit and it does not match with the primary line edit
- The user has finished editing the primary line edit and the confirm line edit is not empty
(The ok button should be disabled immediately in any case as soon as the entries do not match).

This just to avoid that the user is confused because things turn red as soon as he/she starts typing.

Reply:
http://doc.qt.io/qt-5/qlineedit.html#editingFinished

Reply:
Ok, i already implemeneted it. But you can check - it's not work wery well. And i prefer use way without editFinished() signal. And use only passwordChanged() signal + immediate disabling "Ok" button.

I hope, you will agree with me

Reply:
I'd suggest:

    connect(m_pdfExportDialogUi.lineEditPasswordOpen, SIGNAL(textChanged(QString)), this, SLOT(passwordChanged()));
    connect(m_pdfExportDialogUi.lineEditConfirmPasswordOpen, SIGNAL(textChanged(QString)), this, SLOT(passwordChanged()));
    
    /* ... */

    void OutputEditorHOCR::passwordChanged() {
      if(m_pdfExportDialogUi.lineEditPasswordOpen->text() == m_pdfExportDialogUi.lineEditConfirmPasswordOpen->text()){
        m_pdfExportDialogUi.lineEditConfirmPasswordOpen->setStyleSheet(QStringLiteral(""));
        m_pdfExportDialogUi.buttonBox->button(QDialogButtonBox::Ok)->setEnabled(true);
      } else {
        m_pdfExportDialogUi.lineEditConfirmPasswordOpen->setStyleSheet("background: #FF7777; color: #FFFFFF;");
        m_pdfExportDialogUi.buttonBox->button(QDialogButtonBox::Ok)->setEnabled(false);
      }
    }

Reply:
Without additional slots for editingFinished() ?

Reply:
Check current solution, pls.